api/apps/api/src/modules/scenarios/scenarios.controller.ts

```diff
@@ -408,21 +410,43 @@ export class ScenariosController {
     await ensureShapefileHasRequiredFiles(file);
 
     const result = await this.service.processCostSurfaceShapefile(
-      scenarioId,
+      req.params.scenarioId,
```

Comment:
This is weird. `@Param()` decorator does not extract que property from the `request.param` object?
Probably worth checking why, as this is widely used across the API.

Otherwise we should type it in the request, and remove the decorator altogether?

Reply:
I agree - that's exactly what it should do and what it is doing in multiple other endpoints with identical implementation, so I checked many times, tried to remove other decorators (thought maybe some is overriding it), but it just kept as undefined.. 
Maybe it is worth leaving a todo note, to try to solve this mystery in the future, because it doesn't make sence